app/src/main/java/com/babylon/wallet/android/presentation/ui/composables/actionableaddress/ActionableAddressView.kt

```diff
@@ -143,474 +119,373 @@ fun ActionableAddressView(
+            AccompaniedIcon.CopyIcon -> stringResource(id = R.string.addressAction_copyNftId)
+        }
     }
 
-    data class Address(
-        val address: com.radixdlt.sargon.Address
+    @Serializable
+    @SerialName("transaction_id")
+    data class TransactionId(
+        @Serializable(with = IntentHashSerializer::class)
+        val hash: IntentHash,
+        @SerialName("is_visitable_in_dashboard")
+        override val isVisitableInDashboard: Boolean
     ) : ActionableAddress {
 
-        override val copyable: String = address.formatted(format = AddressFormat.RAW)
+        @Transient
+        override val displayable: String = hash.formatted(AddressFormat.DEFAULT)
 
-        override val displayableFull: String = address.formatted(AddressFormat.FULL)
+        @Transient
+        override val icon: AccompaniedIcon = AccompaniedIcon.CopyIcon
 
-        override val displayableTruncated: String = address.formatted(AddressFormat.DEFAULT)
+        @Transient
+        override val action: Action = Action.DropDown(
+            isExpanded = mutableStateOf(false),
+            actions = mutableSetOf<Action.DropDown.ActionItem>(
+                Action.DropDown.ActionItem.Copy(value = rawAddress())
+            ).apply {
+                if (isVisitableInDashboard) {
+                    add(Action.DropDown.ActionItem.Dashboard(url = dashboardUrl()))
+                }
+            }
+        )
 
-        override val dashboardSuffix: String? = when (address) {
-            is com.radixdlt.sargon.Address.AccessController -> null
-            is com.radixdlt.sargon.Address.Account -> "account"
-            is com.radixdlt.sargon.Address.Component -> "component"
-            is com.radixdlt.sargon.Address.Identity -> null
-            is com.radixdlt.sargon.Address.Package -> "package"
-            is com.radixdlt.sargon.Address.Pool -> "pool"
-            is com.radixdlt.sargon.Address.Resource -> "resource"
-            is com.radixdlt.sargon.Address.Validator -> "component"
-            is com.radixdlt.sargon.Address.Vault -> null
-        }?.let {
-            "$it/${address.string.encodeUtf8()}"
+        override fun rawAddress(): String = hash.formatted(format = AddressFormat.RAW)
+
+        override fun truncatedPart(): String = hash.formatted(format = AddressFormat.MIDDLE)
+
+        override fun dashboardUrl(): String = "${hash.networkId.dashboardUrl()}/transaction/${hash.bech32EncodedTxId.encodeUtf8()}"
+
+        @Composable
+        override fun icon() = when (icon) {
+            AccompaniedIcon.CopyIcon -> painterResource(id = R.drawable.ic_copy)
+        }
+
+        @Composable
+        override fun contentDescription() = when (icon) {
+            AccompaniedIcon.CopyIcon -> stringResource(id = R.string.addressAction_copyTransactionId)
         }
     }
 
-    data class GlobalId(
-        val address: NonFungibleGlobalId
-    ) : ActionableAddress {
+    enum class AccompaniedIcon {
```

Comment:
do we need that enum if it has only one case?

Reply:
It is one case for now since the primary action happens to be copy for all addresses that are opening a modal or a popup